Gems/GradientSignal/Code/Source/Components/GradientTransformComponent.cpp

```diff
@@ -453,19 +435,34 @@ namespace GradientSignal
 
         //rebuild bounds from parameters
         m_configuration.m_bounds = m_configuration.m_bounds.GetAbs();
-        m_shapeBounds = AZ::Aabb::CreateFromMinMax(-m_configuration.m_bounds * 0.5f, m_configuration.m_bounds * 0.5f);
+        shapeBounds = AZ::Aabb::CreateFromMinMax(-m_configuration.m_bounds * 0.5f, m_configuration.m_bounds * 0.5f);
 
         //rebuild transform from parameters
         AZ::Matrix3x4 shapeTransformFinal;
         shapeTransformFinal.SetFromEulerDegrees(m_configuration.m_rotate);
         shapeTransformFinal.SetTranslation(m_configuration.m_translate);
         shapeTransformFinal.MultiplyByScale(m_configuration.m_scale);
-        m_shapeTransformInverse = shapeTransformFinal.GetInverseFull();
+        shapeTransformInverse = shapeTransformFinal.GetInverseFull();
 
         // Set everything up on the Gradient Transform
         const bool use3dGradients = m_configuration.m_advancedMode && m_configuration.m_is3d;
         m_gradientTransform = GradientTransform(
-            m_shapeBounds, shapeTransformFinal, use3dGradients, m_configuration.m_frequencyZoom, m_configuration.m_wrappingType);
+            shapeBounds, shapeTransformFinal, use3dGradients, m_configuration.m_frequencyZoom, m_configuration.m_wrappingType);
+
+        // If the transform has changed, send out notifications.
+        if (oldGradientTransform != m_gradientTransform)
+        {
+            // Always notify on the GradientTransformNotificationBus.
+            GradientTransformNotificationBus::Event(
+                GetEntityId(), &GradientTransformNotificationBus::Events::OnGradientTransformChanged, m_gradientTransform);
+
+            // Only notify the DependencyNotificationBus sometimes.
```

Comment:
"sometimes"? :D I'd change it to

```
// Only notify the DependencyNotificationBus when requested by caller.
```

Reply:
lol i got a bit punchy when writing that one. i'll switch to your suggestion. :)